android/canonical/app/src/main/java/com/google/samples/quickstart/canonical/MeFragment.kt

```diff
@@ -37,6 +43,33 @@ class MeFragment : Fragment() {
         return inflater.inflate(R.layout.fragment_me, container, false)
     }
 
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val logoutButton : Button = view.findViewById(R.id.logout_button)
+        logoutButton.setOnClickListener {
+            // Sign out both Google account and Firebase
+            FirebaseAuth.getInstance().signOut()
+            (activity as MainActivity).mGoogleSignInClient.signOut().addOnCompleteListener {
+                val intent = Intent(context, MainActivity::class.java)
+                startActivity(intent)
+            }
+        }
+    }
+
+
+    override fun onStart() {
+        super.onStart()
+        val account = GoogleSignIn.getLastSignedInAccount(this.context)
```

Comment:
Instead of getting last signin account every time, maybe having a viewmodel include the account info so that it can be shared between fragments.

Reply:
In each fragment, GoogleSignIn.getLastSignedInAccount() can be called to get account. The account just need to be initialized once when each fragment is created. 
Under viewModel, the account still need to be initialized(or fetched) from viewModel class. I think they are almost same in workflow.
I am open to viewModel but maybe more comments could be helpful to explain the advantages of viewModel.

Reply:
If sign-in state were handled in the ViewModel you could move some business logic out of the UI and into the ViewModel. You could also observe an account object here and in other fragments. Then if necessary react to a change.

By calling this method repeatedly it may be doing more IPC calls to the library than necessary.

Reply:
I'm not sure the scope of GoogleSignIn.getLastSignedInAccount. My concern is:
User signed in on the phone with account A, when they come to the app, we call this method, returns account A. Then user logged off on their phone, and log in with account B. When they reopen this app, would this method return account B? If so, is that what we expect from this app? 

If I understand correctly, if your viewmodel is scoped to the activity, data can be shared between fragments. So it was initialized when the main activity is launched. Then the account can be consistent between fragments.